sootup.callgraph/src/main/java/sootup/callgraph/AbstractCallGraphAlgorithm.java

```diff
@@ -251,6 +256,61 @@ protected void resolveAllCallsFromSourceMethod(
                                 sourceMethod.getSignature(), targetMethod, stmt, cg, workList)));
   }
 
+  /**
+   * It resolves the start-run implicit calls caused by the given source method
+   *
+   * @param sourceMethod the inspected source method
+   * @param cg implicit start-run calls will be added to the call graph
+   * @param workList new run methods will be added to the work list
+   */
+  protected void implicitStartRunCall(
+      SootMethod sourceMethod, MutableCallGraph cg, Deque<MethodSignature> workList) {
+    ClassType threadType = view.getIdentifierFactory().getClassType("java.lang.Thread");
+    for (Stmt stmt : sourceMethod.getBody().getStmts()) {
+      if (!stmt.isInvokableStmt()) {
+        continue;
+      }
+      AbstractInvokeExpr sourceMethodInvokeExpr =
+          stmt.asInvokableStmt().getInvokeExpr().orElse(null);
+      if (sourceMethodInvokeExpr == null || !sourceMethodInvokeExpr.isJVirtualInvokeExpr()) {
+        continue;
+      }
+      MethodSignature methodSig = sourceMethodInvokeExpr.getMethodSignature();
+      if (!methodSig.getType().equals(VoidType.getInstance())
+          || !methodSig.getParameterTypes().isEmpty()
+          || !methodSig.getName().equals("start")) {
+        continue;
+      }
+      // check if java.lang.Thread is superClass of methodSig.classType()
+      if (typeHierarchy
+          .superClassesOf(methodSig.getDeclClassType())
+          .noneMatch(classType -> classType.equals(threadType))) {
+        continue;
```

Comment:
can you write a test in which sootup finds a fitting start method but the class is no subclass of Thread and therefore there shouldnt be an edge to run().
you have to also write a fitting run method in that testclass